WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostListAdapter.kt

```diff
@@ -20,21 +26,52 @@ import org.wordpress.android.viewmodel.posts.PostListItemType.LoadingItem
 import org.wordpress.android.viewmodel.posts.PostListItemType.PostListItemUiState
 
 private const val VIEW_TYPE_POST = 0
-private const val VIEW_TYPE_ENDLIST_INDICATOR = 1
-private const val VIEW_TYPE_LOADING = 2
+private const val VIEW_TYPE_POST_COMPACT = 1
+private const val VIEW_TYPE_ENDLIST_INDICATOR = 2
+private const val VIEW_TYPE_LOADING = 3
 
 class PostListAdapter(
+    fragment: Fragment,
     context: Context,
     private val postViewHolderConfig: PostViewHolderConfig,
-    private val uiHelpers: UiHelpers
+    private val uiHelpers: UiHelpers,
+    private val postListViewLayoutType: LiveData<PostListViewLayoutType>
 ) : PagedListAdapter<PostListItemType, ViewHolder>(PostListDiffItemCallback) {
     private val layoutInflater: LayoutInflater = LayoutInflater.from(context)
+    private var layout: PostListViewLayoutType
+    private var recyclerView: RecyclerView? = null
+
+    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
+        super.onAttachedToRecyclerView(recyclerView)
+        this.recyclerView = recyclerView
+    }
+
+    override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
+        super.onDetachedFromRecyclerView(recyclerView)
+        this.recyclerView = null
+    }
+
+    init {
+        layout = postListViewLayoutType.value ?: PostListViewLayoutType.defaultValue
+        postListViewLayoutType.observe(fragment, Observer {
```

Comment:
Sure, I was in two minds as to which to do anyway. I was trying to avoid two places having to know about it, but I agree that it doesn't fit in with the way the other property observations work. Updated.